src/socket.cr

```diff
@@ -244,13 +295,8 @@ class Socket < IO::FileDescriptor
   # ```
   def send(message)
     slice = message.to_slice
-    bytes_sent = LibC.send(fd, slice.to_unsafe.as(Void*), slice.size, 0)
-    raise Errno.new("Error sending datagram") if bytes_sent == -1
-    bytes_sent
-  ensure
-    # see IO::FileDescriptor#unbuffered_write
-    if (writers = @writers) && !writers.empty?
-      add_write_event
+    write_syscall_helper(slice, "Error sending message") do |slice|
```

Comment:
I'm not very sure this behaviour is correct, as `write_syscall_helper` retries `send`. This looks correct for stream-type sockets but not for datagram-type sockets, but `send`/`recv` are used for both. However, datagram-type sockets still implement `IO` so i'm not sure what to do. `Socket` in general feels like a messy part of the stdlib to me.